File: kotlin/src/main/kotlin/exams/exams2/Exam102.kt
package exams.exams2

fun giveGem(gem: IntArray, operations: Array<IntArray>): Int {
    for ((x, y) in operations) {
        var n = gem[x] / 2
        gem[x] -= n
        gem[y] += n
    }

    return gem.max() - gem.min()
}

File: python/src/exams/exams2/Exam102.py
""" 102-宝石补给

"""

from typing import List


def giveGem(gem: List[int], operations: List[List[int]]) -> int:
    for x, y in operations:
        n = gem[x] // 2
        gem[x] -= n
        gem[y] += n

    return max(gem) - min(gem)
